sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -242,6 +245,32 @@ public async ValueTask WaitForEnvironmentAsync()
                 }
                 await task;
             }
+
+            if (!GlobalIsRunningInCI &&
+                Mode is RecordedTestMode.Live or RecordedTestMode.Record &&
+                !s_initializedEnvironmentTypes.Contains(_type))
+            {
+                s_initializedEnvironmentTypes.Add(_type);
+
+                string resourceGroup = GetOptionalVariable("RESOURCE_GROUP");
+                if (resourceGroup == null)
+                {
+                    return;
+                }
+
+                ArmClient armClient = new ArmClient(Credential);
+                ResourceGroup rg = await armClient.DefaultSubscription.GetResourceGroups().GetAsync(resourceGroup);
```

Comment:
`DefaultSubscription` seems a bit risky, given that we don't know what subscriptions the credential may have access to.    Should we consider using `SubscriptionId` to explicitly request a subscription when creating the ARM client?   _([ref](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs#L95))_

Reply:
We should pull the sub id from the New-TestResources output https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/New-TestResources.ps1#L520

Reply:
Filed https://github.com/Azure/azure-sdk-for-net/issues/23906